### PR TITLE
[New Algorithm] - Triangular Numbers

### PR DESCRIPTION
**Pull request by Tauseef-Hilal**
### Describe your change:
New algorithm for generating Triangular numbers.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by Tauseef-Hilal** [comment]
Hi @ChrisO345 
Please review this

File: maths/triangular_numbers.py
Comment (tianyizheng02) [inline maths/triangular_numbers.py:1]: Please move this file into `maths/special_numbers`

File: maths/triangular_numbers.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+A triangular number or triangle number counts objects arranged in an
+equilateral triangle. This module provides a function to generate
+first `n` triangular numbers.
+
+For more information about triangular numbers, refer to:
+https://en.wikipedia.org/wiki/Triangular_number
+"""
+
+
+def generate_triangular_numbers(count: int) -> list[int]:
```
Comment (tianyizheng02) [inline maths/triangular_numbers.py:11]: Why generate the triangular numbers in a list? Why not just return the nth triangular number? The triangular numbers have a closed-form formula, after all: https://en.wikipedia.org/wiki/Triangular_number

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
We already have [maths/special_numbers/polygonal_numbers.py](https://github.com/TheAlgorithms/Python/blob/a9cee1d933606092eb966a601eb1d9efd6e054af/maths/special_numbers/polygonal_numbers.py), which is technically a more general case than just the triangular numbers. However, because the triangular numbers specifically are so well-known, I feel that it's OK to include this PR as a special case.

File: maths/triangular_numbers.py (hunk 1 above)
Comment (Tauseef-Hilal) [reply maths/triangular_numbers.py:11]: Okay, I'll modify the function to calculate nth triangular number

**Comment by Tauseef-Hilal** [comment]
@tianyizheng02 Done. Please review the changes

File: maths/special_numbers/triangular_numbers.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    Examples:
+    >>> nth_triangular_number(1)
+    1
+    >>> nth_triangular_number(3)
+    6
+    >>> nth_triangular_number(5)
+    15
+    """
+    if position < 1:
+        raise ValueError("param `position` must be greater than 0")
```
Comment (tianyizheng02) [inline maths/special_numbers/triangular_numbers.py:30]: 1. The triangular numbers are also defined for n = 0 (see Wikipedia)
2. Could you add a doctest for this ValueError case?

File: maths/special_numbers/triangular_numbers.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+A triangular number or triangle number counts objects arranged in an
+equilateral triangle. This module provides a function to generate n'th
+triangular number.
+
+For more information about triangular numbers, refer to:
+https://en.wikipedia.org/wiki/Triangular_number
+"""
+
+
+def nth_triangular_number(position: int) -> int:
```
Comment (tianyizheng02) [inline maths/special_numbers/triangular_numbers.py:11]:
```suggestion
def triangular_number(position: int) -> int:
```
For simplicity

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

File: maths/special_numbers/triangular_numbers.py (hunk 2 above)
Comment (Tauseef-Hilal) [reply maths/special_numbers/triangular_numbers.py:30]: Okay

**Review by tianyizheng02 — APPROVED** [review]

**Pull request merged**